install_protoc_from_source.sh

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env sh
+
+PROTOC_VERSION="2.6.1"
+
+if [[ ! -d "$HOME/protobuf/lib" ||  "$(protoc --version)" != *${PROTOC_VERSION}* ]]; then
```

Comment:
Luckily $HOME/protobuf/lib is a non-standard install point, so if someone wanted another version of protobuf installed via conventional methods, it would likely result in another binary being placed in /usr/bin or whatever. At that point, one could just provide a FQ path, or do some environment wrangling to invoke the version they want.